src/mcode-pilot/utils/FilterOptions.js

```diff
@@ -0,0 +1,85 @@
+export default class FilterOptions {
+    constructor(props) {
+        this.filters = props;
+        this.filterList = this.getFiltersAsList();
+    }
+
+    getAllActiveFilters() {
+        return this.filterList.filter((filter) => {
+            return filter.selected;
+        });
+    }
+
+    getFiltersAsList() {
+        let returnArray = [];
```

Comment:
[DP2] Related to DP1, can we explicitly overwrite the value/push onto/gradually concatenate returnArray each time we call recursiveFilterSearch for a given key? Let's not depend on sideeffects of a function call to mutate returnArray, if possible.